code/modules/mob/living/simple_animal/friendly/farm_animals.dm

```diff
@@ -339,3 +339,59 @@
 	name = "emperor penguin"
 	desc = "Emperor of all he surveys."
 
+/mob/living/simple_animal/pig
+	name = "pig"
+	desc = "Known for being the main ingrediant in bacon."
+	icon_state = "pig"
+	icon_living = "pig"
+	icon_dead = "pig_dead"
+	speak = list("OINKKK?","Oink!","RRRRRREEEEEEEEEE!")
+	speak_emote = list("Oinks","Oinks hauntingly")
+	emote_hear = list("squeels")
+	emote_see = list("shakes its head")
+	speak_chance = 1
+	turns_per_move = 5
+	see_in_dark = 6
+	meat_type = /obj/item/weapon/reagent_containers/food/snacks/meat/pig
+	meat_amount = 15 //Pigs are fat as fuck, as they grow this will increase
+	response_help  = "pets"
+	response_disarm = "gently pushes aside"
+	response_harm   = "kicks"
+	attacktext = "kicked"
+	health = 280
+	autoseek_food = 0
+	beg_for_food = 0
+	var/datum/reagents/udder = null
+	var/amount_grown = 0
+	mob_size = 20//based on mass of holstein fresian dairy cattle, what the sprite is based on
+
+/mob/living/simple_animal/pig/Initialize()
+	. = ..()
+	var/gender = pick(MALE, FEMALE)
+
+	if(gender == FEMALE) // You cant milk boy pigs you pervs
+		udder = new(50)

```

Comment:
50u sounds like way too much considering pigs can only "milk" for around 2.5% of the time a cow can.